sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -49,8 +49,7 @@ private[hive] case class MetastoreRelation(
   override def equals(other: Any): Boolean = other match {
     case relation: MetastoreRelation =>
       databaseName == relation.databaseName &&
-        tableName == relation.tableName &&
-        output == relation.output
+        tableName == relation.tableName
```

Comment:
When we change `equals`, we also need to change `hashCode`. 

`LogicalRelation`'s [`equals`](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala#L56) also checks the output. 

When the underlying Hive tables change the schema, are we able to detect it?